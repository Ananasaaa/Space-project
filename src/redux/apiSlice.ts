import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const nasaApi = createApi({
  reducerPath: 'nasaApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'https://api.nasa.gov/',
  }),
  endpoints: (builder) => ({
    getSpaceImage: builder.query({
      query: () => ({
        url: 'planetary/apod',
        params: {
          api_key: process.env.REACT_APP_NASA_KEY as string,
        },
      }),
    }),
    getMarsPhotos: builder.query({
      query: ({ sol = 4000, camera, page = 1 }) => ({
        url: 'mars-photos/api/v1/rovers/curiosity/photos',
        params: {
          sol,
          page,
          ...(camera ? { camera } : {}),
          api_key: process.env.REACT_APP_NASA_KEY as string,
        },
      }),
    }),
  }),
});

export const { useGetSpaceImageQuery, useGetMarsPhotosQuery } = nasaApi;
