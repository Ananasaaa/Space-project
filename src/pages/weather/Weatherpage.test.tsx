import { render } from '@testing-library/react';
import Weatherpage from './index';
import { useGetMarsWeatherQuery } from '../../redux/apiSlice';

jest.mock('../../redux/apiSlice');

describe('Weatherpage component', () => {
  it('should render weather cards correctly', () => {
    (useGetMarsWeatherQuery as jest.Mock).mockReturnValue({
      data: {
        sol_keys: ['1000'],
        '1000': {
          First_UTC: '2024-04-25T12:00:00Z',
          AT: { av: -65, mn: -80, mx: -50 },
          PRE: { av: 750 },
          WD: { most_common: { compass_point: 'N' } },
        },
      },
      isLoading: false,
      isError: false,
    });

    const { asFragment } = render(<Weatherpage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
