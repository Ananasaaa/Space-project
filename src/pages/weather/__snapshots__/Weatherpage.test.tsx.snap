// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weatherpage component should render weather cards correctly 1`] = `
<DocumentFragment>
  <div
    class="weatherpage"
    style="background-image: url(/img/bg_weatherpage.jpg); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <div
      class="weather-container"
    >
      <h1
        class="page-title"
      >
        Weather on Mars
      </h1>
      <button
        class="unit-toggle"
      >
        Switch to 째F
      </button>
      <div
        class="weather-cards"
      >
        <div
          class="weather-card fade-in"
        >
          <h2
            class="sol"
          >
            Sol 1000
          </h2>
          <p
            class="date"
          >
            4/25/2024
          </p>
          <div
            class="temperature"
          >
            <p
              class="temp-high"
            >
              Max: -50째C
            </p>
            <p
              class="temp-low"
            >
              Min: -80째C
            </p>
            <p
              class="temp-avg"
            >
              Average: -65째C
            </p>
          </div>
          <div
            class="pressure"
          >
            <p>
              Pressure: 750 Pa
            </p>
          </div>
          <div
            class="windspeed"
          >
            <p>
              Wind direction: N
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
